docs/cloud/satellites/self-hosted.md

```diff
@@ -0,0 +1,215 @@
+{% endhint %}
+
+### Auto-discovery of Host _\*experimental\*_
+
+When `SATELLITE_HOST` is left unset, the satellite will attempt to auto-discover its host address. This feature may not always work, and currently only supports environments using cloud-init (such as EC2).
+
+## Connecting to a Self-Hosted Satellite
+
+Once the satellite has finished registering, it can be selected and managed using typical earthly satellite commands from the CLI. 
+For example, to select the satellite for use in subsequent builds:
+
+```
+earthly sat select my-satellite
+```
+
+Or, to invoke a build once on the satellite, without persistently selecting it:
+
+```
+earthly --sat my-satellite +my-target
+```
+
+Note that `earthly-v0.8.0` or later is required to connect to a self-hosted satellite.
+
+## Managing Self-Hosted Satellites
+
+A list of satellites in your org be viewed via the satellite ls command. For example:
+
+```
+> earthly sat ls
+NAME          PLATFORM     SIZE         VERSION  STATE
+my-satellite  linux/amd64  self-hosted  v0.8.3   operational
+...
+```
+
+The size field of a self-hosted satellite will be listed as `self-hosted`, as opposed to one of the sizes of Earthly Cloud’s managed satellites.
+
+The state field will be either `operational` or `offline`. A self-hosted satellite automatically transitions to an `offline` state when it is gracefully terminated, or if no heartbeat has been received by Earthly’s servers for a while.
+
+You can run `satellite rm` on a self-hosted satellite when it is in an offline state to remove it from your account permenantly.
+
+## Platform-Specific Guides
+
+### Kubernetes
+
+Below is a basic example of how to start a self-hosted satellite as a Kubernetes Pod:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-satellite
+  namespace: earthly-satellites
+spec:
+  containers:
+   - name:
+   valueFrom:
+     fieldRef:
+       fieldPath: metadata.name
+      image: earthly/satellite:v0.8.3
+      securityContext:
+        privileged: true
+      ports:
+        - containerPort: 8372
+       env:
+        - name: EARTHLY_ORG
+          value: my-org
+        - name: EARTHLY_TOKEN
+          value: u4a*************l92
+        - name: SATELLITE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: SATELLITE_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+```
+
+Note that this example uses the Pod’s IP address (via [Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/)) as the `SATELLITE_HOST` value. The Earthly CLI must be able to reach the IP on its network.
+
+If the Earthly CLI should connect via a different address, such as DNS, then this value should be provided as the `SATELLITE_HOST` instead.
+
+
+### AWS EC2
```

Comment:
I wonder if the EC2 version should be more encouraged in this guide since VMs provide a better security sandbox for builds. Just a thought.

Reply:
People seem very inclined to use K8S, but what if we put a "recommended" label on AWS , and/or bump it higher in the doc?

Reply:
SGTM

Reply:
That makes sense to me too. EC2 should be above k8s and encouraged.